### PR TITLE
add quadruple input hatch

### PR DESCRIPTION
**Pull request by GlodBlock**
add an input hatch that can hold 4 different types of fluid

### CODE REVIEW

File: src/main/java/gregtech/api/gui/GT_Container_2by2_Fluid.java [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+                tTank.setDrainableStack(GT_Utility.getFluidFromDisplayStack(tTank.getStackInSlot(2)));
+            }
+            GT_MetaTileEntity_Hatch_MultiInput tTank = (GT_MetaTileEntity_Hatch_MultiInput) mTileEntity.getMetaTileEntity();
+            GT_Container_2by2_Fluid.IFluidAccess tDrainableAccess = GT_Container_2by2_Fluid.IFluidAccess.from(tTank, aSlotIndex);
+            return handleFluidSlotClick(tDrainableAccess, aPlayer, aMouseclick == 0, true, !tTank.isDrainableStackSeparate(), aSlotIndex);
+        }
+        return super.slotClick(aSlotIndex, aMouseclick, aShifthold, aPlayer);
+    }
+
+    protected static ItemStack handleFluidSlotClick(GT_Container_2by2_Fluid.IFluidAccess aFluidAccess, EntityPlayer aPlayer, boolean aProcessFullStack, boolean aCanDrain, boolean aCanFill, int aSlot) {
```
Comment (Glease) [inline src/main/java/gregtech/api/gui/GT_Container_2by2_Fluid.java:51]: Just pull this method up to GT_Container. Duplicating code everywhere make it hard to maintain the codebase.

**Review by Glease — APPROVED** [review]
I'm still curious as why you need a 4 tank input hatch, but the code looks ok.

**Comment by GTNHAfx237v7** [comment]
Does this work with assembly line?

**Comment by Glease** [comment]
Probably doesn't. Too much code just assumed one fluid per hatch.

**Pull request merged**